_aggregations/metric/scripted-metric.md

```diff
@@ -55,18 +231,38 @@ GET opensearch_dashboards_sample_data_logs/_search
 ```
 {% include copy-curl.html %}
 
-#### Example response
+
+The response returns three values in the `value` object, demonstrating how a scripted metric can return multiple metrics at once by using a map in the state:
 
 ```json
-...
-"aggregations" : {
-  "responses.counts" : {
-    "value" : {
-      "other" : 0,
-      "success" : 12832,
-      "error" : 1242
+{
+  ...
+  "hits": {
+    "total": {
+      "value": 5,
+      "relation": "eq"
+    },
+    "max_score": null,
+    "hits": []
+  },
+  "aggregations": {
+    "responses_by_type": {
+      "value": {
+        "other": 1,
+        "success": 2,
+        "error": 2
+      }
     }
   }
- }
 }
 ```
+
+## Handling empty buckets (no documents scenario)
+
+When using a `scripted_metric` aggregation as a subaggregation within a bucket aggregation (such as `terms`), it is important to account for buckets that contain no documents on certain shards. In such cases, those shards return a `null` value for the aggregation state. During the `reduce_script` phase, the `states` array may therefore include `null` entries corresponding to these shards. To ensure reliable execution, the `reduce_script` must be designed to handle `null` values gracefully. A common approach is to include a conditional check, such as `if (state != null)`, before accessing or operating on each state. Failure to implement such checks can result in runtime errors when processing empty buckets across shards.
+
+
+## Performance considerations
+
+Because scripted metrics run custom code for every document and possibly accumulate large in-memory state, they can be slower than built-in aggregations. The intermediate state from each shard has to be serialized to send it to the coordinating node. Therefore if your `state` is very large, it can consume a lot of memory and network bandwidth. To keep your searches efficient, make your scripts as lightweight as possible and avoid accumulating unnecessary data in the `state`. Use the combine stage to shrink state data before sending, as demonstrated in [Calculating net profit from transactions](#calculating-net-profit-from-transactions), and only collect the values that you truly need to produce the final metric.
+
```

Comment:
Line 267: This paragraph contains both "state" and `state`. Should they all be `state`? Also, in the last sentence, should `combine` be in code font (it is above)?

Reply:
@natebower I've updated all states to `state` and same for `map`, this one is difficult to gauge, can you have another look through this to make sure it all makes sense?